src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -711,19 +744,14 @@ public Node call() throws Exception {
                                                 template,
                                                 vmName,
                                                 deploymentName);
-                                    } catch (Exception e) {
+                                    } catch (AzureCloudException e) {
```

Comment:
Why change to AzureCloudException? Should other exceptions also be handled here?